pmd-java/src/main/resources/category/java/bestpractices.xml

```diff
@@ -1633,4 +1633,41 @@ public class Foo {
         </example>
     </rule>
 
+    <rule name="WhileLoopWithLiteralBoolean"
```

Comment:
just my 2 cents here… I got here to a rule named `WhileLoopWithLiteralBoolean`, so I was expecting a rule that would detect and warn on unconditional loops… but I see this rule actually allows `while (true)`.

Although I understand the reasoning behind this rule as it stands (readability / dead code removal), I can't help but wonder…

1. isn't there a better name for this?
2. dead-code removal, shouldn't be in a separate rule detecting all control statements? (`if (false)`, etc.)
3. if other than dead-code removal this aims at readability, should it be in bestpractices or as a codestyle rule?

Reply:
The name of the rule is quite neutral IMHO. E.g. it doesn't start with "Avoid...". So, it just a rule about While Loops with Literal Booleans... I agree, that you cannot decide from the name, whether you would want to use the rule - you'll need to check the rule description.

About the dead-code removal: I'd suggest to leave it as is and revisit this rule when we have other rules in the dead-code-removal space.